hunting/utils.py

```diff
@@ -0,0 +1,134 @@
+    """Load the hunting index.yml file."""
+    index_file = HUNTING_DIR / "index.yml"
+    if not index_file.exists():
+        click.echo(f"No index.yml found at {index_file}.")
+        return {}
+
+    with open(index_file, 'r') as f:
+        hunting_index = yaml.safe_load(f)
+
+    return hunting_index
+
+
+def load_toml(source: Union[Path, str]) -> Hunt:
+    """Load and validate TOML content as Hunt dataclass."""
+    if isinstance(source, Path):
+        if not source.is_file():
+            raise FileNotFoundError(f"TOML file not found: {source}")
+        contents = source.read_text(encoding="utf-8")
+    else:
+        contents = source  # Assuming it's a TOML string
+
+    toml_dict = tomllib.loads(contents)
+
+    # Validate and load the content into the Hunt dataclass
+    return Hunt(**toml_dict["hunt"])
+
+
+def load_all_toml(base_path: Path):
+    """Load all TOML files from the directory and return a list of Hunt configurations and their paths."""
+    hunts = []
+    for toml_file in base_path.rglob("*.toml"):
+        hunt_config = load_toml(toml_file)
+        hunts.append((hunt_config, toml_file))
+    return hunts
+
+
+def save_index_file(base_path: Path, directories: dict) -> None:
+    """Save the updated index.yml file."""
+    index_file = base_path / "index.yml"
+    with open(index_file, 'w') as f:
+        yaml.safe_dump(directories, f, default_flow_style=False, sort_keys=False)
+    print(f"Index YAML updated at: {index_file}")
+
+
+def validate_link(link: str):
+    """Validate and return the link."""
+    http = urllib3.PoolManager()
+    response = http.request('GET', link)
+    if response.status != 200:
+        raise ValueError(f"Invalid link: {link}")
+
+
+def update_index_yml(base_path: Path) -> None:
+    """Update index.yml based on the current TOML files."""
+    directories = load_index_file()
+
+    # Load all TOML files recursively
+    toml_files = base_path.rglob("queries/*.toml")
+
+    for toml_file in toml_files:
+        # Load TOML and extract hunt configuration
+        hunt_config = load_toml(toml_file)
+
+        folder_name = toml_file.parent.parent.name
+        uuid = hunt_config.uuid
+
+        entry = {
+            'name': hunt_config.name,
+            'path': f"./{toml_file.relative_to(base_path).as_posix()}",
+            'mitre': hunt_config.mitre
+        }
+
+        # Check if the folder_name exists and if it's a list, convert it to a dictionary
+        if folder_name not in directories:
+            directories[folder_name] = {uuid: entry}
+        else:
+            if isinstance(directories[folder_name], list):
+                # Convert the list to a dictionary, using UUIDs as keys
+                directories[folder_name] = {item['uuid']: item for item in directories[folder_name]}
+            directories[folder_name][uuid] = entry
+
+    # Save the updated index.yml
+    save_index_file(base_path, directories)
+
+
+def filter_elasticsearch_params(config: dict) -> dict:
```

Comment:
We cast anything from the config file to `detection_rules.misc.get_elasticsearch_client`. In doing this, I accidentally left the `KIBANA_URL` in there from a previous config. As such it threw an error as that was for the kibana client. Therefore, rather than require users to adjust existing configs, I settled with filtering with what is defined as acceptable parameters of `get_elasticsearch_client`. Happy to adjust if need be.

Reply:
make sense.